imap_processing/ultra/l1c/make_helio_index_maps.py

```diff
@@ -0,0 +1,335 @@
+    num_steps: int,
+    start_et: float,
+    instrument_frame: SpiceFrame = SpiceFrame.IMAP_ULTRA_90,
+    compute_bsf: bool = False,
+    boundary_points: int = 8,
+) -> xr.Dataset:
+    """
+    Create HEALPix index maps for heliocentric observations.
+
+    This function generates exposure maps that account for spacecraft
+    velocity aberration, multiple energy bins, and spin phase sampling.
+
+    Parameters
+    ----------
+    nside : int
+        HEALPix nside parameter (determines angular resolution).
+    spin_duration : float
+        Total spin period in seconds.
+    num_steps : int
+        Number of spin phase steps to sample.
+    start_et : float
+        Start ephemeris time.
+    instrument_frame : SpiceFrame, optional
+        SpiceFrame of the instrument (default IMAP_ULTRA_90).
+    compute_bsf : bool, optional
+        If True, compute boundary scale factors (default False).
+    boundary_points : int, optional
+        Number of boundary points to sample per pixel (default 8).
+
+    Returns
+    -------
+    xarray.Dataset
+        Dataset with dimensions (step, energy, pixel) containing index,
+        theta, phi, and bsf data variables, plus ra and dec coordinates.
+    """
+    # Get spacecraft velocity at start time
+    state = imap_state(start_et, ref_frame=SpiceFrame.IMAP_DPS, observer=SpiceBody.SUN)
+    sc_vel = state[3:6]  # Extract [vx, vy, vz]
+
+    logger.info("Spacecraft velocity: %s km/s", sc_vel)
+    logger.info("Speed: %.2f km/s", np.linalg.norm(sc_vel))
+
+    # Build energy bins
+    _, energy_midpoints, energy_bin_geometric_means = build_energy_bins()
+    num_energy_bins = len(energy_bin_geometric_means)
+
+    # Get number of pixels
+    npix = hp.nside2npix(nside)
+
+    # Compute RA/Dec for pixel centers
+    pixel_indices = np.arange(npix)
+
+    # Time parameters
+    end_et = start_et + spin_duration
+    dt_step = spin_duration / num_steps
+
+    # Pre-compute all pixel vectors once
+    pixel_vecs = np.array(hp.pix2vec(nside, pixel_indices, nest=False)).T  # (npix, 3)
+
+    # Initialize output arrays
+    index_map = np.zeros((num_steps, num_energy_bins, npix))
+    theta_map = np.zeros((num_steps, num_energy_bins, npix))
+    phi_map = np.zeros((num_steps, num_energy_bins, npix))
+    bsf_map = np.zeros((num_steps, num_energy_bins, npix))
+
+    logger.info(
+        "Processing %d time steps, %d energy bins, %d pixels...",
+        num_steps,
+        num_energy_bins,
+        npix,
+    )
+    if compute_bsf:
+        logger.info(
+            "Computing boundary scale factors with %d points per pixel",
+            boundary_points,
+        )
+    # TODO vectorize loop
+    time_id = 0
+    t = start_et
+    while t < (end_et - dt_step / 2):
+        # Get rotation matrix for this time step
+        rotation_matrix = get_rotation_matrix(
+            np.array([t]),
+            from_frame=instrument_frame,
+            to_frame=SpiceFrame.IMAP_DPS,
+        )[0]
```

Comment:
Instead of passing an array of length one for the `et` argument and taking the zeroth element of the result, I think you should be able to do:
```suggestion
        rotation_matrix = get_rotation_matrix(
            t,
            from_frame=instrument_frame,
            to_frame=SpiceFrame.IMAP_DPS,
        )
```